web/src/beta/lib/core/engines/Cesium/useEngineRef.ts

```diff
@@ -505,7 +506,96 @@ export default function useEngineRef(
       findFeaturesByIds: (layerId: string, featureIds: string[]): Feature[] | undefined => {
         const viewer = cesium.current?.cesiumElement;
         if (!viewer || viewer.isDestroyed()) return;
-        return featureIds.map(f => e.findFeatureById(layerId, f)).filter((v): v is Feature => !!v);
+        return findFeaturesFromLayer(viewer, layerId, featureIds, e => {
+          const f = convertObjToComputedFeature(viewer, e)?.[1];
+          return f
+            ? ({
+                ...f,
+                type: "feature",
+              } as Feature)
+            : undefined;
+        });
+      },
+      findComputedFeatureById: (
+        layerId: string,
+        featureId: string,
+      ): ComputedFeature | undefined => {
+        const viewer = cesium.current?.cesiumElement;
+        if (!viewer || viewer.isDestroyed()) return;
+        const entity = findEntity(viewer, layerId, featureId);
+        const tag = getTag(entity);
+        if (!tag?.featureId) {
+          return;
+        }
+        if (entity instanceof Cesium.Entity) {
+          // TODO: Return description for CZML
+          return (
+            tag.computedFeature ?? {
+              type: "computedFeature",
+              id: tag.featureId,
+              properties: entity.properties,
+            }
+          );
+        }
+        if (entity instanceof Cesium.Cesium3DTileFeature) {
+          return (
+            tag.computedFeature ?? {
+              type: "computedFeature",
+              id: tag.featureId,
+              properties: Object.fromEntries(
+                entity.getPropertyIds().map(key => [key, entity.getProperty(key)]),
+              ),
+            }
+          );
+        }
+        return;
+      },
+      findComputedFeaturesByIds: (
+        layerId: string,
+        featureIds: string[],
+      ): ComputedFeature[] | undefined => {
+        const viewer = cesium.current?.cesiumElement;
+        if (!viewer || viewer.isDestroyed()) return;
+        return findFeaturesFromLayer(viewer, layerId, featureIds, e => {
+          return convertObjToComputedFeature(viewer, e)?.[1];
+        });
```

Comment:
```suggestion
        return findFeaturesFromLayer(viewer, layerId, featureIds, e => convertObjToComputedFeature(viewer, e)?.[1]);
```
Since you’re using arrow functions and implicit returns, consider adopting a more functional style for simple transformations. I might be worth doing the same for the rest of places where this practice can be adopted.